src/forms/form-fields/drop-down-async.js

```diff
@@ -36,8 +37,10 @@ class DropDownAsync extends Component {
             });
         }
 
-        //referenceType updated, reload
-        if (this.props.referenceType !== newProps.referenceType || this.props.queryParamFilter !== newProps.queryParamFilter) {
+        // referenceType updated, reload
+        // Deep equal queryParamFilter, as this may have a new reference each render.
+        // Should not have that many elems.
+        if (this.props.referenceType !== newProps.referenceType || !isEqual(newProps.queryParamFilter, this.props.queryParamFilter)) {
```

Comment:
As discussed on slack, you could opt for this faster but less pretty alternative to `isEqual`:
`JSON.stringify(newProps.queryParamFilter) !== JSON.stringify(this.props.queryParamFilter)`

Reply:
Actually.... Just had another idea: You mentioned on slack that `queryParamFilter` is just an array. So before you start doing expensive object comparisons, you could just check the length first. If these don't match you know to refresh, and only if the lengths match you start doing the object comparison....